### PR TITLE
Default field annotation doesn't change existent column #1758

### PR DESCRIPTION
**Pull request by m3hm3t**
Fixed #1753

### CODE REVIEW

File: dialect/sql/schema/migrate.go [hunk 1]
```diff
@@ -375,6 +375,9 @@ func (m *Migrate) changeSet(curr, new *Table) (*changes, error) {
 		// Change nullability of a column.
 		case c1.Nullable != c2.Nullable:
 			change.column.modify = append(change.column.modify, c1)
+		// Change default value
```
Comment (a8m) [inline dialect/sql/schema/migrate.go:378]:
```suggestion
		// Change default value.
```

File: dialect/sql/schema/postgres.go [hunk 2]
```diff
@@ -415,9 +415,30 @@ func (d *Postgres) alterColumn(c *Column) (ops []*sql.ColumnBuilder) {
 	} else {
 		ops = append(ops, b.Column(c.Name).Attr("SET NOT NULL"))
 	}
+
+	if c.supportDefault() && c.Default != nil {
+		ops = append(ops, d.writeSetDefault(b.Column(c.Name), c))
+	}
+
 	return ops
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:423]:
```suggestion
	if c.Default != nil && c.supportDefault() {
		ops = append(ops, d.writeSetDefault(b.Column(c.Name), c))
	}
	return ops
```

File: dialect/sql/schema/postgres.go [hunk 3]
```diff
@@ -415,9 +415,30 @@ func (d *Postgres) alterColumn(c *Column) (ops []*sql.ColumnBuilder) {
...
+	case bool:
+		attr = strconv.FormatBool(v)
+	case string:
+		if t := c.Type; t != field.TypeUUID && t != field.TypeTime && !t.Numeric() {
+			// Escape single quote by replacing each with 2.
+			attr = fmt.Sprintf("'%s'", strings.ReplaceAll(v, "'", "''"))
+		}
+	}
+	b.Attr("SET DEFAULT " + attr)
+
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:438]:
```suggestion
```

Comment (a8m) [reply dialect/sql/schema/postgres.go:438]: m.. this is a copy of `Column.defaultValue`, don't you prefer to move it to shared function?

Comment (m3hm3t) [reply dialect/sql/schema/postgres.go]: I have changed `Column.defaultValue` to eliminate duplicated code lines. 

I hope it is better.

Comment (m3hm3t) [reply dialect/sql/schema/postgres.go]: I have updated the method to use `Postgres.writeDefault`.

File: dialect/sql/schema/postgres.go [hunk 4]
```diff
@@ -415,9 +420,18 @@ func (d *Postgres) alterColumn(c *Column) (ops []*sql.ColumnBuilder) {
...
 	}
+
+	if c.Default != nil && c.supportDefault() {
+		ops = append(ops, d.writeSetDefault(b.Column(c.Name), c))
+	}
 	return ops
 }
 
+func (d *Postgres) writeSetDefault(b *sql.ColumnBuilder, c *Column) *sql.ColumnBuilder {
+	d.writeDefault(b, c, "SET DEFAULT ")
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:431]:
```suggestion
	d.writeDefault(b, c, "SET DEFAULT")
```

File: dialect/sql/schema/postgres.go [hunk 5]
```diff
@@ -415,9 +420,18 @@ func (d *Postgres) alterColumn(c *Column) (ops []*sql.ColumnBuilder) {
 	} else {
 		ops = append(ops, b.Column(c.Name).Attr("SET NOT NULL"))
 	}
+
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:423]:
```suggestion
```

File: dialect/sql/schema/postgres.go [hunk 6]
```diff
@@ -403,7 +403,12 @@ func (d *Postgres) writeDefault(b *sql.ColumnBuilder, c *Column) {
 			attr = fmt.Sprintf("'%s'", strings.ReplaceAll(v, "'", "''"))
 		}
 	}
-	b.Attr("DEFAULT " + attr)
+
+	var sb strings.Builder
+	sb.WriteString(clause)
+	sb.WriteString(attr)
+
+	b.Attr(sb.String())
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:411]:
```suggestion
	b.Attr(caluse + " " + attr)
```

string builder is not needed here.

File: dialect/sql/schema/postgres.go [hunk 7]
```diff
@@ -380,7 +380,7 @@ func (d *Postgres) addColumn(c *Column) *sql.ColumnBuilder {
 		b.Attr("GENERATED BY DEFAULT AS IDENTITY")
 	}
 	c.nullable(b)
-	d.writeDefault(b, c)
+	d.writeDefault(b, c, "DEFAULT ")
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:383]:
```suggestion
	d.writeDefault(b, c, "DEFAULT")
```

**Review by a8m — APPROVED** [review]
Amazing work @m3hm3t.

File: dialect/sql/schema/postgres.go [hunk 8]
```diff
@@ -403,7 +403,8 @@ func (d *Postgres) writeDefault(b *sql.ColumnBuilder, c *Column) {
 			attr = fmt.Sprintf("'%s'", strings.ReplaceAll(v, "'", "''"))
 		}
 	}
-	b.Attr("DEFAULT " + attr)
+
+	b.Attr(caluse + " " + attr)
```
Comment (a8m) [inline dialect/sql/schema/postgres.go:407]:
```suggestion
	b.Attr(clause + " " + attr)
```

**Comment by a8m** [comment]
@all-contributors please add @m3hm3t for code

**Pull request merged**